### PR TITLE
Add support for custom prefixes (GH-number)

### LINKED ISSUE #7: Link issues that take the form of "GH-<number>"
Essentially, the same thing that I requested in the Refined Github repository, here: https://github.com/sindresorhus/refined-github/issues/3262

Was asked to open it here, so here we are :)

### PR DESCRIPTION
**Pull request by fregante**
- Fixes https://github.com/sindresorhus/linkify-issues/issues/7

Blocked by:

- [x] #18 
- [x] https://github.com/sindresorhus/issue-regex/pull/20

### CODE REVIEW

**Comment by fregante** [comment]
This required a bit of reorganization of the code because there's no longer a single global regex, but it depends on the input. Since the regex is [reused](https://github.com/sindresorhus/linkify-issues/pull/18#discussion_r1775115721), this meant having to pass around this regex. The code was ugly.

Also I realized we don't really need to re-parse the reference again: the captured groups are available via both `.replace()` and `.split()`. The latter requires a bit of mental gymnastics because:

- there's no `.groups`
- `.split()` generates a `Array<string>` that needs to be looped in a certain way to extract this information

File: index.js [hunk 1]
```diff
@@ -42,22 +49,46 @@ const linkify = (reference, options) => {
...
+		fragment.append(parts[index]);
+
+		const reference = parts[index + 1];
+		const groups = {
+			organization: parts[index + 2],
+			repository: parts[index + 3],
+			issueNumber: parts[index + 4],
+		};
+
+		fragment.append(domify(linkify(reference, groups, options)));
```
Comment (fregante) [inline index.js:91]: If you don't like this, I can replace it with:

```suggestion
		fragment.append(domify(linkify(reference, regex.exec(reference).groups, options)));
```

As for the loop itself, I find this `for` loop to be more readable than the previous version, including the `parts` explanation above.

File: index.d.ts [hunk 2]
```diff
@@ -22,6 +22,13 @@ export type Options = {
 	@default 'https://github.com'
 	*/
 	readonly baseUrl?: string;
+
+	/**
+	Additional reference prefix to support. It can be set to `undefined` to disable the default.
```
Comment (sindresorhus) [inline index.d.ts:27]: Not sure the point of supporting `undefined` to disable default? What will it even do then?

File: index.d.ts [hunk 3]
```diff
@@ -22,6 +22,13 @@ export type Options = {
...
+
+	@default 'GH-'
+	*/
+	readonly additionalPrefix?: string | undefined;
```
Comment (sindresorhus) [inline index.d.ts:49]: Needs to be added to the readme.

File: index.js (hunk 1 above)
Comment (sindresorhus) [reply index.js:91]: The current version is fine.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Ready for release after merging this?

File: index.d.ts [hunk 4]
```diff
@@ -28,13 +28,15 @@ export type Options = {
 
  	@default 'GH-'
 
+	@example
  	```
```
Comment (fregante) [inline index.d.ts:32]: Are .d.ts files excluded from XO? There are a couple of `<space><tab>` instances here

Comment (sindresorhus) [reply index.d.ts:32]: Strange. Will look into it.

**Pull request merged**